### PR TITLE
OD-113 [Fix] User will be redirected to the currect folder when restore file and press 'Go to folder' button

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-113

## Description
We have created new functions that are controlling user redirection in the file manager. For the basis of these functions was taken from the callbacks that were used on click and dblclick events in the file manager.

## Screenshots/screencasts
https://share.getcloudapp.com/RBu40oyE

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko @ivandevupp

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -96,11 +96,53 @@ function parseThumbnail(file) {
...
+  };
+  backItem.type = 'folderId';
+
+  navStack.push(backItem);
+
+  updatePaths();
+  getFolderContents($item);
+}
+
 function navigateToFolder(item) {
```
Comment (tonytlwu) [inline js/interface.js:134]:
```suggestion
function navigateToFolderItem(item) {
```

File: js/interface.js [hunk 2]
```diff
@@ -589,7 +631,7 @@ function getFolderContents(el, isRootFolder) {
   if (el.attr('data-type') === 'trash') {
     getTrashFilesData(filterFiles, filterFolders);
```
Comment (tonytlwu) [inline js/interface.js:632]: Don't need `else`

```suggestion
    return getTrashFilesData(filterFiles, filterFolders);
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js [hunk 3]
```diff
@@ -96,11 +96,53 @@ function parseThumbnail(file) {
...
+
+  navStack.push(backItem);
+
+  updatePaths();
+  getFolderContents($item);
+}
+
+function navigateToFolderItem(item) {
+  if (item.data('folder')) {
+    navigateToRootLevelFolder({ appId: item.data('app-id'), orgId: item.data('org-id') })
```
Comment (tonytlwu) [inline js/interface.js:140]: I'd rather you only pass `appId` or `orgId`, depending on how you want the `navigateToRootFolder` function to behave.

The `navigateToRootFolder` function can also check to make sure that it prioritises `appId`. This way, the caller is clear in its intentions and the function is able to handle the scenarios too.

File: js/interface.js (hunk 2 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu added suggestion

File: js/interface.js (hunk 3 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu I've added a suggestion and I hope I understand you correctly with that suggestion.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Does not meet coding standard

**Comment by YaroslavOvdii** [comment]
@tonytlwu Fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Medium
Affected: Studo users managing files
Verify: Users can navigate between folders

Squash and merge: Go to current folder after restoring file

**Pull request merged**